projects/plugins/jetpack/extensions/blocks/contact-form/util/with-style-variables.js

```diff
@@ -0,0 +1,10 @@
+import './form-styles.js';
+import { useCallback } from '@wordpress/element';
+
+export const withStyleVariables = WrappedComponent => props => {
+	const EDITOR_SELECTOR = '[data-type="jetpack/contact-form"]';
+	/* eslint-disable-next-line no-undef */
+	const generateVariables = useCallback( () => generateStyleVariables( EDITOR_SELECTOR ), [] );
+
+	return <WrappedComponent style={ generateVariables() } { ...props } />;
```

Comment:
I'm also wondering if we can optimize this so we're not re-rendering the entire probe on every render 🤔. 
Doing it on every render would certainly pick up on any changes quicker than otherwise but it doesn't guarantee that either.

Reply:
Both work, `useMemo` is particularly useful if you're doing an expensive calculation that you might have to run every now and then if something changes, but don't want to run it every time the component renders.  

In this case, `useEffect` would ensure it only runs once.